be/src/exec/pipeline/exchange/sink_buffer.h

```diff
@@ -50,34 +58,76 @@ class SinkBuffer {
     }
 
     ~SinkBuffer() {
-        _pending_chunks.shutdown();
+        _is_finished = true;
+        _cache_empty_cv.notify_one();
         _thread.join();
-        if (_closure->unref()) {
-            delete _closure;
+        for (auto* closure : _closures) {
+            if (closure->unref()) {
+                delete closure;
+            }
         }
     }
 
-    void add_request(const TransmitChunkInfo& request) { _pending_chunks.put(request); }
+    void add_request(const TransmitChunkInfo& request) {
+        std::lock_guard<std::mutex> l(_mutex);
+        _caches[request.channel_id].push(request);
+        _cache_empty_cv.notify_one();
+    }
 
     void process() {
         try {
             MemTracker* prev_tracker = tls_thread_status.set_mem_tracker(_mem_tracker);
             DeferOp op([&] { tls_thread_status.set_mem_tracker(prev_tracker); });
 
-            while (true) {
-                if (_closure->has_in_flight_rpc()) {
-                    std::this_thread::sleep_for(std::chrono::milliseconds(1));
-                    continue;
+            while (!_is_finished) {
+                {
+                    std::unique_lock<std::mutex> l(_mutex);
+                    bool is_cache_empty = true;
+                    for (auto& cache : _caches) {
+                        if (!cache.empty()) {
+                            is_cache_empty = false;
+                            break;
+                        }
+                    }
+                    if (is_cache_empty) {
+                        _cache_empty_cv.wait(l);
+                    }
                 }
 
-                TransmitChunkInfo info;
-                if (!_pending_chunks.blocking_get(&info)) {
-                    break;
+                const size_t spin_threshould = 100;
+                size_t spin_iter = 0;
+
+                for (; spin_iter < spin_threshould; ++spin_iter) {
+                    bool find_any = false;
+                    for (auto& cache : _caches) {
+                        if (cache.empty()) {
+                            continue;
+                        }
+
+                        // std::queue' read is concurrent safe without mutex
+                        if (!_closures[cache.front().channel_id]->has_in_flight_rpc()) {
+                            TransmitChunkInfo info = cache.front();
+                            find_any = true;
+                            _send_rpc(info);
+                            {
+                                std::lock_guard<std::mutex> l(_mutex);
+                                cache.pop();
+                            }
+                            info.params->release_finst_id();
```

Comment:
deallocate  info.params->release_finst_id() . finst_id in Info copies has  memory leak